src/com/nutrons/nu17/subsystems/Drivetrain.java

```diff
@@ -93,11 +108,110 @@ public void resetEncoder() {
    * Resets the gyro.
    */
   public void resetGyro() {
-    this.headingGyro.reset();
+    this.gyro.reset();
 
   }
 
+  public void setPercentDrive() {
+    this.frontLeft.changeControlMode(TalonControlMode.PercentVbus);
+    this.frontRight.changeControlMode(TalonControlMode.PercentVbus);
+  }
+
+  public void disableBreakMode() {
+    this.frontLeft.enableBrakeMode(false);
+    this.frontRight.enableBrakeMode(false);
+  }
+
   public void initDefaultCommand() {
     drive.tankDrive(OperatorInterface.DRIVER_PAD.getY(), OperatorInterface.DRIVER_PAD.getX());
   }
+
+  // PID
+  // TODO: tune these constants
+  private static final double P_HEADING = 0;
+  private static final double I_HEADING = 0;
+  private static final double D_HEADING = 0;
+
+  // TODO: tune these constants
+  private static final double P_TURN = 0;
+  private static final double I_TURN = 0;
+  private static final double D_TURN = 0;
+
+  // TODO: tune these constants
+  private static final double P_DISTANCE = 0;
+  private static final double I_DISTANCE = 0;
+  private static final double D_DISTANCE = 0;
+
+  private double heading;
+  public final PIDController holdHeading = new PIDController(P_HEADING, I_HEADING, D_HEADING,
+      new GyroWrapper(), new HoldHeadingOutput());
+  private double angle;
+  public final PIDController turnToAngle =
+      new PIDController(P_TURN, I_TURN, D_TURN, new GyroWrapper(), new TurnToAngleOutput());
+  private double distance;
+  public final PIDController driveDistance = new PIDController(P_DISTANCE, I_DISTANCE, D_DISTANCE,
+      new EncoderWrapper(), new DriveDistanceOutput());
+
+  private class GyroWrapper implements PIDSource {
```

Comment:
Move this into separate file or explain why second one is needed.  If they serve the same function, create a common interface